### ISSUE TITLE
Error after upgrading from version 3.11.12 -> 3.11.13

### ISSUE THREAD

**Issue by sevenseacat** [team-alembic/ash_authentication#433]
Not sure where this is coming from, but it was picked up by Dependabot - our app no longer compiled on CI.

```
== Compilation error in file lib/my_app/accounts/user.ex ==
** (RuntimeError) Exception in transformer AshAuthentication.Strategy.Custom.Transformer on MyApp.Accounts.User:

expected a map, got: nil
    (elixir 1.15.5) lib/map.ex:486: Map.take/2
    (ash_authentication 3.11.13) lib/ash_authentication/strategies/password/transformer.ex:78: anonymous fn/2 in AshAuthentication.Strategy.Password.Transformer.transform/2
    (ash_authentication 3.11.13) lib/ash_authentication/strategies/password/transformer.ex:75: AshAuthentication.Strategy.Password.Transformer.transform/2
    (ash_authentication 3.11.13) lib/ash_authentication/strategies/custom/transformer.ex:90: AshAuthentication.Strategy.Custom.Transformer.do_transform/3
    (ash_authentication 3.11.13) lib/ash_authentication/strategies/custom/transformer.ex:44: anonymous fn/2 in AshAuthentication.Strategy.Custom.Transformer.do_strategy_transforms/1
    (elixir 1.15.5) lib/enum.ex:4830: Enumerable.List.reduce/3
    (elixir 1.15.5) lib/enum.ex:2564: Enum.reduce_while/3
    /Users/me/Projects/work/my-app/lib/my_app/accounts/user.ex:1: (file)
```

**Comment by zachdaniel** [comment]
I just pushed something to `main` that ought to resolve this. It will at least handle the `nil` value, but there may be a deeper issue.

**Comment by jimsynz** [comment]
Should be fixed in v3.11.15.  Can you confirm @sevenseacat?

**Comment by sevenseacat** [comment]
We're running 3.11.15 now and it's all good!
